cmd/server-main.go

```diff
@@ -439,6 +439,10 @@ func initServer(ctx context.Context, newObject ObjectLayer) error {
 				// These messages only meant primarily for distributed setup, so only log during distributed setup.
 				logger.Info("All MinIO sub-systems initialized successfully in %s", time.Since(t1))
 			}
+
+			// Initialize bucket notification system.
+			logger.LogIf(GlobalContext, globalEventNotifier.InitBucketTargets(GlobalContext, newObject))
```

Comment:
@harshavardhana here, we should also call 

```go
// List buckets to initialize bucket metadata sub-sys.
buckets, err := newObject.ListBuckets(GlobalContext, BucketOptions{})
if err != nil {
	logger.LogIf(GlobalContext, fmt.Errorf("Unable to list buckets to initialize bucket metadata sub-system: %w", err))
}

// Initialize bucket metadata sub-system.
globalBucketMetadataSys.Init(GlobalContext, buckets, newObject)
```

otherwise, the `bucketRulesMap` will be empty and we will miss the events.